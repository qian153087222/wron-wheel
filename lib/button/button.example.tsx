import React from 'react'
import Button from './button';

const ButtonExamople: React.FunctionComponent = () => {
    return (
    <div>
        <Button Type='Primary' name={'Primary'} />
        <Button Type='Default' name={'Default'} />
        <Button Type='Dashed' name={'Dashed'} />
        <Button Type='Danger' name={'Danger'} />
    </div>)

}

export default ButtonExamople;
