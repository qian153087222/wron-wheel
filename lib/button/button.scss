.wron-button{
    padding: 5px 12px;
    font-size: 12px;
    color: white;
    border-radius: 4px;
    border: none;
    &.wron-Primary{
        background: #108EE9;
        &:hover{
            background: #49a9ee;
        }
        &:focus{
            outline:0;
        }
        &:active{
            background: #0e77ca;
        }
    }
    &.wron-Default{
        color: #000;
        background: #fff;
        border: 1px #D9D9D9 solid;
        &:hover{
            color: #0e77ca;
            border: 1px #0e77ca solid;
        }
        &:focus{
            outline:0;
        }
        &:active{
            color: #0e77ca;
            border: 1px #0e77ca solid;
        }
    }
    &.wron-Dashed{
        color: #000;
        background: #fff;
        border: 1px #D9D9D9 dashed;
        &:hover{
            color: #0e77ca;
            border: 1px #0e77ca dashed;
        }
        &:focus{
            outline:0;
        }
        &:active{
            color: #0e77ca;
            border: 1px #0e77ca dashed;
        }
    }
    &.wron-Danger{
        background: #fff;
        border: 1px #D9D9D9 solid;
        color: #F04134;
        &:hover{
            background: #F04134;
            color: #fff;
        }
        &:focus{
            outline:0;
        }
        &:active{
            background: #d73435;
        }
    }
    
}