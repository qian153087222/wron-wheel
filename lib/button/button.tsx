//有时候import不能正确引入，导致cannot read property createElement of undefined错误，就要自己重命名
import *as React from 'react';
import './button.scss';
import classes from '../helper/classes'

interface ButtonProps extends React.ButtonHTMLAttributes<Element> {
    name: string;
    Type:string;
    className?:string
}

const Button: React.FunctionComponent<ButtonProps> = ({ className, Type,name, ...restProps }) => {

    return (
        <button className={classes(`wron-button wron-${Type}`, className) } {...restProps}>{name}</button>
    )
};

export default Button;







