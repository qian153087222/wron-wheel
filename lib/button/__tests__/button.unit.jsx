import renderer from 'react-test-renderer';
import React from 'react';
import button from '../button';
//Enzyme是由Airbnb开源的一个React的JavaScript测试工具
/**
 * enzyme支持三种方式的渲染：
shallow：浅渲染，是对官方的Shallow Renderer的封装。将组件渲染成虚拟DOM对象，只
会渲染第一层，子组件将不会被渲染出来，因而效率非常高。不需要DOM环境， 并可以使用jQuery的方式访问组件的信息；
render：静态渲染，它将React组件渲染成静态的HTML字符串，然后使用Cheerio这个库解析这段
字符串，并返回一个Cheerio的实例对象，可以用来分析组件的html结构。
mount：完全渲染，它将组件渲染加载成一个真实的DOM节点，用来测试DOM API的交互和组件
的生命周期，用到了jsdom来模拟浏览器环境。
 */
import { mount } from 'enzyme';

describe('button', () => {
    it('测试buttonui', () => {
        const json = renderer.create(<button name='Primary' />).toJSON();
        //在当前目录生成一个快照，生成__snapshots__文件,对的保存
        expect(json).toMatchSnapshot();
    })
    it('onClick', () => {
        let n = 1;
        const fn = () => {
            n = n + 1;
        };
        const component = mount(<button name='Primary' onClick={fn} />);
        //simulate模拟
        component.find('button').simulate('click');
        expect(n).toEqual(2);
    })
})
