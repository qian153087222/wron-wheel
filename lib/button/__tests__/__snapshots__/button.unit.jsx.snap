// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 测试buttonui 1`] = `
<button
  name="Primary"
/>
`;
